libs/sdk-ui-kit/src/@ui/UiMenu/hooks.ts

```diff
@@ -0,0 +1,117 @@
+// (C) 2022-2025 GoodData Corporation
+
+import React from "react";
+import { IUiMenuContext, UiMenuProps } from "./types.js";
+import { makeMenuKeyboardNavigation } from "../@utils/keyboardNavigation.js";
+import { getClosestFocusableItemId, getItemParent, getSubMenuSiblingItems } from "./itemUtils.js";
+
+export function useItemRefs(focusedId?: string) {
+    const itemRefs = React.useRef<{ [id: string]: HTMLElement }>({});
+
+    const setItemRef = React.useCallback(
+        (id: string) => (element: HTMLElement | null) => {
+            if (!element) {
+                delete itemRefs.current[id];
+                return;
+            }
+
+            if (focusedId === id) {
+                element.scrollIntoView({ block: "nearest", behavior: "smooth" });
+            }
+
+            itemRefs.current[id] = element;
+        },
+        [focusedId],
+    );
+
+    return { itemRefs, setItemRef };
+}
+
+export function useKeyNavigation<InteractiveItemData, StaticItemData>({
+    menuContextRef,
+    shouldKeyboardActionPreventDefault,
+    shouldKeyboardActionStopPropagation,
+    onUnhandledKeyDown,
+}: {
+    menuContextRef: React.MutableRefObject<IUiMenuContext<InteractiveItemData, StaticItemData>>;
+    onUnhandledKeyDown: UiMenuProps<InteractiveItemData, StaticItemData>["onUnhandledKeyDown"];
+    shouldKeyboardActionPreventDefault?: boolean;
+    shouldKeyboardActionStopPropagation?: boolean;
+}) {
+    return makeMenuKeyboardNavigation({
+        shouldPreventDefault: shouldKeyboardActionPreventDefault,
+        shouldStopPropagation: shouldKeyboardActionStopPropagation,
+
+        onFocusPrevious: () => {
+            const { setFocusedId, items, isItemFocusable } = menuContextRef.current;
+
+            setFocusedId(
+                (prevId) =>
+                    getClosestFocusableItemId({
```

Comment:
did I get it correct that next/prev does not support cycling from last to first menu item and vice versa?